File: SmartReceipts/Reports/ReceiptIndexer.m
//
//  ReceiptIndexer.m
//  SmartReceipts
//
//  Created by Jaanus Siim on 25/04/15.
//  Copyright (c) 2015 Will Baumann. All rights reserved.
//

#import "ReceiptIndexer.h"
#import "WBReceipt.h"

@implementation ReceiptIndexer

+ (NSArray *)indexReceipts:(NSArray *)receipts filteredWith:(BOOL (^)(WBReceipt *))filter {
    NSMutableArray *filteredAndIndexed = [NSMutableArray array];
    int i = 0;
    for (WBReceipt *rec in receipts) {
        if (filter(rec)) {
            continue;
        }
        ++i;
        [rec setReportIndex:i];
        [filteredAndIndexed addObject:rec];
    }
    return filteredAndIndexed.copy;
}

@end

File: SmartReceipts/Reports/ReceiptIndexer.swift
//
//  ReceiptIndexer.swift
//  SmartReceipts
//
//  Created by Bogdan Evsenev on 03/12/2017.
//  Copyright © 2017 Will Baumann. All rights reserved.
//

import Foundation

class ReceiptIndexer {
    class func indexReceipts(_ receipts: [WBReceipt], filter: (WBReceipt) -> Bool) -> [WBReceipt] {
        var result = [WBReceipt]()
        var i = 1
        for receipt in receipts {
            if filter(receipt) {
                continue
            }
            receipt.reportIndex = i
            result.append(receipt)
            i += 1
        }
        return result
    }
}
